src/scripts/main.js

```
@@ -1,3 +1,91 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const tBody = document.querySelector('tbody');
+  const rowsArray = Array.from(tBody?.rows ?? []);
+
+  if (!table || !tBody || !rowsArray.length) {
+    return;
+  }
+
+  const tableFieldsType = ['string', 'string', 'number', 'number'];
+
+  table.addEventListener('click', (e) => {
+    const th = e.target.closest('th');
+
+    if (!th) {
+      return;
+    }
+
+    const columnIndex = th.cellIndex;
+
+    sortColumn(columnIndex, tableFieldsType[columnIndex]);
+  });
+
+  function sortColumn(index, type) {
+    let copmareFun;
+
+    switch (type) {
+      case 'number':
+        copmareFun = function (trA, trB) {
+          if (checkTD(trA.cells[index], trB.cells[index])) {
+            return 0;
+          }
+
+          const numberA = convertToNumber(trA.cells[index].textContent);
+          const numberB = convertToNumber(trB.cells[index].textContent);
+
+          if (checkNumber(numberA, numberB)) {
+            return 0;
+          }
+
+          return numberA - numberB;
+        };
+        break;
+      case 'string':
+        copmareFun = function (trA, trB) {
+          if (checkTD(trA.cells[index], trB.cells[index])) {
+            return 0;
+          }
+
+          const stringA = trA.cells[index].textContent.trim();
+          const stringB = trB.cells[index].textContent.trim();
+
+          return stringA.localeCompare(stringB);
+        };
+        break;
+      default:
+        return;
```

The `default` case in the `switch` (lines 58–59) silently returns, which can hide misconfiguration (unknown column type). Either validate the `type` earlier and bail with a clear message or fall back to a sensible default (e.g., treat as `string`). Explicit handling makes behavior deterministic and easier to test.